src/codeu/chat/client/commandline/Chat.java

```diff
@@ -136,8 +148,11 @@ public void invoke(Scanner args) {
     //
     panel.register("u-add", new Panel.Command() {
       @Override
-      public void invoke(Scanner args) {
-        final String name = args.hasNext() ? args.nextLine().trim() : "";
+      public void invoke(List<String> args) {
+        //TODO: possible place of error, not sure how to test username creation
+        // what's in the List being passed in? Is it already tokenized? The scanner code took the entire line?
```

Comment:
If the `List<String>` passed in contains the output of `.handleCommand()` then it should just be a list of tokens from a String that's already been tokenized, so I was hoping that's the case. The issue with passing a `Tokenizer` into each method is that there needs to be a String passed into `Tokenizer` and the input `String line` is passed into `.handleCommand()` but not saved as a variable for other methods to use.